crates/precompile/src/bls12_381/g2_add.rs

```diff
@@ -26,12 +27,15 @@ pub(super) fn g2_add(input: &Bytes, gas_limit: u64) -> PrecompileResult {
         )));
     }
 
+    let [a_x_0, a_x_1, a_y_0, a_y_1] = remove_g2_padding(&input[..PADDED_G2_LENGTH])?;
+    let [b_x_0, b_x_1, b_y_0, b_y_1] = remove_g2_padding(&input[PADDED_G2_LENGTH..])?;
+
```

Comment:
Most of the changes is removing the padding at the precompile level versus the crypto wrapper level -- so the backend only knows about "correctly formatted" points instead of the EVM specific padding that gets added

Reply:
This makes it simpler